### PR TITLE
add source language for translation

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: ApiBundle/Controller/ContentController.php [hunk 1]
```diff
@@ -32,17 +33,44 @@ class ContentController extends BaseController
...
-        $contentRepository = $this->get('open_orchestra_model.repository.content');
-        $content = $contentRepository->findOneByContentIdAndLanguageAndVersion($contentId, $language, $version);
+        $content = $this->findOneContent($contentId, $language, $request->get('version'));
 
         if (!$content) {
-            $defaultCurrentSiteLanguage = $this->get('open_orchestra_backoffice.context_manager')->getCurrentSiteDefaultLanguage();
-            $oldContent = $contentRepository->findOneByContentIdAndLanguageAndVersion($contentId, $defaultCurrentSiteLanguage);
+            $sourceLanguage = $request->get('source_language');
+            if (!$sourceLanguage) {
+                throw new ContentNotFoundHttpException();
```
Comment (itkg-ngilain) [inline ApiBundle/Controller/ContentController.php]: Maybe a new exception : the problem is not that the content is not found but there's a missing parameter

File: ApiBundle/Controller/ContentController.php [hunk 2]
```diff
@@ -32,17 +33,44 @@ class ContentController extends BaseController
...
+        }
+
+        return $this->get('open_orchestra_api.transformer_manager')->get('content')->transform($content);
+    }
+
+    /**
+     * @param Request $request
+     * @param string  $contentId
+     *
+     * @Config\Route("/{contentId}/or/create", name="open_orchestra_api_content_show_or_create")
```
Comment (itkg-ngilain) [inline ApiBundle/Controller/ContentController.php]: I find this route strange, I find something like /show-or-create/{contentId} more conventional

Comment (nicolasThal) [inline ApiBundle/Controller/ContentController.php]: I would rather choose /{contentId}/show-or-create, this will be more restfull

**Comment by nicolasThal** [comment]
@itkg-ngilain correction done

File: ApiBundle/Controller/ContentController.php [hunk 3]
```diff
@@ -32,17 +34,44 @@ class ContentController extends BaseController
...
-        $contentRepository = $this->get('open_orchestra_model.repository.content');
-        $content = $contentRepository->findOneByContentIdAndLanguageAndVersion($contentId, $language, $version);
+        $content = $this->findOneContent($contentId, $language, $request->get('version'));
 
         if (!$content) {
-            $defaultCurrentSiteLanguage = $this->get('open_orchestra_backoffice.context_manager')->getCurrentSiteDefaultLanguage();
-            $oldContent = $contentRepository->findOneByContentIdAndLanguageAndVersion($contentId, $defaultCurrentSiteLanguage);
+            $sourceLanguage = $request->get('source_language');
+            if (!$sourceLanguage) {
+                throw new ContentSourceNotFoundHttpException();
```
Comment (itkg-ngilain) [inline ApiBundle/Controller/ContentController.php]: I think the exception is still not correct : we can not say that the content is not found as we haven't search for it

Comment (nicolasThal) [inline ApiBundle/Controller/ContentController.php]: My bad, is SourceLanguageNotFound ok for you?

**Comment by itkg-ngilain** [comment]
:+1:

**Pull request merged**